### ISSUE TITLE
pydantic.error_wrappers.ValidationError: 6 validation errors for ParsingModel[PaginatedFileList]

### ISSUE THREAD

**Issue by hwuiwon** [merge-api/merge-python-client#38]
Hi, I'm getting `pydantic.error_wrappers.ValidationError` when using the code below. It used to work before so I'm guessing that there has been some change in serverside json response, which is not reflected in `merge-python-client`. Can you guys take a look at it and let me know? Thanks

Code
```python
from merge.client import Merge

client = Merge(api_key=MERGE_API_KEY, account_token=ACCOUNT_TOKEN)

res = client.filestorage.files.list(page_size=100, cursor=None)

print(res)
```

Error
```bash
Traceback (most recent call last):
  File "/Users/hwuiwon/Desktop/test.py", line 5, in <module>
    res = client.filestorage.files.list(page_size=100, cursor=None)
  File "/usr/local/anaconda3/envs/prism/lib/python3.10/site-packages/merge/resources/filestorage/resources/files/client.py", line 106, in list
    return pydantic.parse_obj_as(PaginatedFileList, _response.json())  # type: ignore
  File "pydantic/tools.py", line 38, in pydantic.tools.parse_obj_as
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 6 validation errors for ParsingModel[PaginatedFileList]
__root__ -> results -> 0 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 1 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 2 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 3 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 4 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 5 -> permissions -> 0
  str type expected (type=type_error.str)
```

**Comment by dsinghvi** [comment]
@rmkonnur https://github.com/merge-api/merge-python-client/blob/42f42e4649032d3428065ca41e8f76a309e4a2ee/src/merge/resources/filestorage/types/file.py#L31-L33 it looks like `permissions` should be typed `str | List[str]` in the OpenAPI spec

**Comment by hwuiwon** [comment]
When can I expect the fix?

**Comment by rmkonnur** [comment]
You can expect a fix by EoD today. Will update in this thread when its pushed out.

**Comment by hwuiwon** [comment]
Should I create our own version of merge client that sends request directly to the API endpoint? I personally believe that not able to list files is a P1 issue and should be fixed ASAP. This has been blocking the service for few days now.

**Comment by dsinghvi** [comment]
@hwuiwon one option to unblock yourself is to make this change to the library locally (https://github.com/merge-api/merge-python-client/pull/39). Can you make that change and confirm that the SDK works end-to-end? 

Apologize for any inconvenience.

**Comment by hwuiwon** [comment]
@dsinghvi I just tried and that would yield a different error.

```bash
12 validation errors for ParsingModel[PaginatedFileList]
__root__ -> results -> 0 -> permissions
  str type expected (type=type_error.str)
__root__ -> results -> 0 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 1 -> permissions
  str type expected (type=type_error.str)
__root__ -> results -> 1 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 2 -> permissions
  str type expected (type=type_error.str)
__root__ -> results -> 2 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 3 -> permissions
  str type expected (type=type_error.str)
__root__ -> results -> 3 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 4 -> permissions
  str type expected (type=type_error.str)
__root__ -> results -> 4 -> permissions -> 0
  str type expected (type=type_error.str)
__root__ -> results -> 5 -> permissions
  str type expected (type=type_error.str)
__root__ -> results -> 5 -> permissions -> 0
  str type expected (type=type_error.str)
```

**Comment by hwuiwon** [comment]
@dsinghvi 

You should change the type to `typing.Optional[typing.List[typing.Dict[str, typing.Any]]]`

If it's okay, I'll fork it and open a new PR addressing this issue

**Comment by dsinghvi** [comment]
Got it -- that makes sense, the root cause is that the OpenAPI spec is incorrectly representing the API. So @rmkonnur is planning to fix the OpenAPI generation which will then regenerate a proper client library.

**Comment by lucasgadams** [comment]
Im encountering this problem. It seems that previously the permissions field was a list of (str) ids, like:
```
  "permissions": [
    "1496fd39-1b31-42be-9aa4",
    "99b321d0-5dad-4268-b73c",
    ]
 ```
 
 But now the permission object looks like:
<img width="618" alt="image" src="https://github.com/merge-api/merge-python-client/assets/36425095/1e85f897-60aa-438f-966e-f5b8238c47b0">


So i guess just changing the type to str | JSON would fix the library.. but its a kind of breaking change

**Comment by rmkonnur** [comment]
Hi @hwuiwon, I pushed an update of the File Storage OpenAPI spec to generate the SDK this morning. Please try it out to confirm it fixes your issue. Apologies in the delay here - feel free to escalate via Intercom in the bottom right hand corner of the Merge dashboard for higher priority issues. 
Thanks
